Mathlib/Logic/Function/HasFiniteFibers.lean

```diff
@@ -28,25 +28,65 @@ of fibers. -/
 def HasFiniteFibers (f : α → β) : Prop :=
   ∀ ⦃S : Set β⦄, S.Finite → (f ⁻¹' S).Finite
 
-variable {f : α → β}
+/-- `Function.HasFiniteFibers f` means that the preimage by `f` of any finite set is finite. We
+show in `Function.hasFiniteFibers_iff_finite_fibers` that this is actually equivalent to finiteness
+of fibers. -/
+def HasFiniteFibersOn (f : α → β) (U : Set α) : Prop :=
+  ∀ ⦃S : Set β⦄, S.Finite → (f ⁻¹' S ∩ U).Finite
```

Comment:
Why do you not take `Function.hasFiniteFibers_iff_finite_fibers` as the definition? This is the easiest one to show in concrete cases, I would think.

Reply:
You're probably right. I was enjoying writing `hf hS` all the time (where `hS : S.Finite), but that's not important at all.